### PR TITLE
Use existing method for WuiDom creation

### PR DESCRIPTION
**Pull request by micky2be**
Use existing method on WuiDom instead of depending on createHtmlElement.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -103,22 +91,23 @@ WuiDom.prototype._assign = function (tagName, options) {
...
-	if (options && options.hidden) {
-		// start hidden
+	// start hidden
+	if (options.hidden) {
 		this.hide();
 	}
 
-	return this.rootElement;
+	// set identifier (used by getChild)
+	this._name = String(options.name);
```
Comment (ronkorving) [inline index.js]: And if options.name is undefined, you want _name to become `'undefined'`?

File: index.js [hunk 2]
```diff
@@ -339,7 +328,7 @@ WuiDom.prototype.getChildren = function () {
  * @returns {WuiDom|undefined}
  */
 WuiDom.prototype.getChild = function (childName) {
-	return this._childrenMap[childName];
+	return this._childrenMap[String(childName)];
```
Comment (ronkorving) [inline index.js]: String casting is done for you automatically, so what's the purpose of this?

File: index.js [hunk 3]
```diff
@@ -161,7 +150,7 @@ WuiDom.prototype.removeChild = function (child) {
 
 	this.rootElement.removeChild(child.rootElement);
 	this._childrenList.splice(siblingIndex, 1);
-	if (this._childrenMap.hasOwnProperty(child._name)) {
+	if (child._name in this._childrenMap) {
```
Comment (ronkorving) [inline index.js]: Why the change?

File: index.js [hunk 4]
```diff
@@ -103,22 +91,23 @@ WuiDom.prototype._assign = function (tagName, options) {
...
+	if (options.hidden) {
 		this.hide();
 	}
 
-	return this.rootElement;
+	// set identifier (used by getChild)
+	this._name = String(options.name);
+
+	this.addClassNames(options.className);
+	this.setStyles(options.style || {});
```
Comment (ronkorving) [inline index.js]: Why call functions that are going to do absolutely nothing? Why all this deoptimization?

File: index.js (hunk 2 above)
Comment (micky2be) [inline index.js]: weird, I though I removed that cast

File: index.js (hunk 3 above)
Comment (micky2be) [inline index.js]: readability

File: index.js (hunk 1 above)
Comment (micky2be) [inline index.js]: Yes and No. (I forgot to push the latest change)
I let the developer take the responsibility to not set `name` to undefined

File: index.js (hunk 4 above)
Comment (micky2be) [inline index.js]: I can add an if statement indeed

File: index.js (hunk 3 above)
Comment (ronkorving) [inline index.js]: The `in` operator traverses the prototype chain, something you may want to keep in mind.

Comment (micky2be) [inline index.js]: We talk about a literal objet here (a map), so no worries.

File: index.js [hunk 5]
```diff
@@ -103,22 +91,30 @@ WuiDom.prototype._assign = function (tagName, options) {
...
-		// start hidden
+	// start hidden
+	if (options.hidden) {
 		this.hide();
 	}
 
-	return this.rootElement;
+	// set identifier (used by getChild)
+	if ('name' in options) {
+		this._name = String(options.name);
```
Comment (micky2be) [inline index.js]: @ronkorving To come back to previous comment on this line.
'name' can still be set to `undefined` but like previously stated, I let the responsibility to the developer.

File: index.js (hunk 3 above)
Comment (ronkorving) [inline index.js]:
```js
console.log(({}).__proto__);  // < an object
```

File: index.js (hunk 5 above)
Comment (ronkorving) [inline index.js]: I think before there was no check of the name property actually being set. Now there is, so it's cool.

File: index.js (hunk 3 above)
Comment (micky2be) [inline index.js]: Right. So if the name of my child is "__proto__" it will return `true`.
But in that case we are talking about a different problem here, no?
